stdlib/public/Distributed/DistributedActor.swift

```diff
@@ -363,9 +363,13 @@ extension DistributedActor {
   ///
   /// When the actor is remote, the closure won't be executed and this function will return nil.
   @_alwaysEmitIntoClient
-  // we need to silgen_name here because the signature is the same as __abi_whenLocal,
+  // we need to @abi here because the signature is the same as __abi_whenLocal,
   // and even though this is @AEIC, the symbol name would conflict.
-  @_silgen_name("$s11Distributed0A5ActorPAAE20whenLocalTypedThrowsyqd__Sgqd__xYiYaYbqd_0_YKXEYaqd_0_YKs8SendableRd__s5ErrorRd_0_r0_lF")
+  @abi(
+    public nonisolated func __api_whenLocal<T: Sendable, E>(
```

Comment:
Probably this would be nicer if it stayed using the explicit "explaining why this method is different" name like we did in the silgen name:
```suggestion
    public nonisolated func whenLocalTypedThrows<T: Sendable, E>(
```

Reply:
There's actually three different variants. Here's what I think I'll do here:

```
  @abi(
    // We need to @abi here because the signature is the same as
    // `__separately_compiled_typed_throws_whenLocal(_:)`, and even though this
    // is @AEIC, the symbol name would conflict.
    public nonisolated func __typed_throws_whenLocal<T: Sendable, E>(
      _ body: @Sendable (isolated Self) async throws(E) -> T
    ) async throws(E) -> T?
  )
  @_alwaysEmitIntoClient
  public nonisolated func whenLocal<T: Sendable, E>(
    _ body: @Sendable (isolated Self) async throws(E) -> T
  ) async throws(E) -> T? { /* snip */ }

  // ABI: This is a workaround when in Swift 6 this method was introduced
  // in order to support typed-throws, but missed to add @_aeic.
  // In practice, this method should not ever be used by anyone, ever.
  @abi(
    public nonisolated func whenLocal<T: Sendable, E>(
      _ body: @Sendable (isolated Self) async throws(E) -> T
    ) async throws(E) -> T?
  )
  @usableFromInline
  nonisolated func __separately_compiled_typed_throws_whenLocal<T: Sendable, E>(
    _ body: @Sendable (isolated Self) async throws(E) -> T
  ) async throws(E) -> T? { /* snip */ }

  // ABI: Historical whenLocal, rethrows was changed to typed throws `throws(E)`
  @abi(
    public nonisolated func whenLocal<T: Sendable>(
      _ body: @Sendable (isolated Self) async throws -> T
    ) async rethrows -> T?
  )
  @usableFromInline
  nonisolated func __rethrows_whenLocal<T: Sendable>(
    _ body: @Sendable (isolated Self) async throws -> T
  ) async rethrows -> T? { /* snip */ }
```

Reply:
This looks good, thank you @beccadax! I'm really enjoying this way to spell these